### PR TITLE
ARCHI-245 implements sdk tests for http-get

### PR DESCRIPTION
**Pull request by guillaumelamirand**
## Issue

https://gravitee.atlassian.net/browse/ARCHI-245

## Description

Add new integrations using sdk for http-get with kafka and mqtt

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-fxerfboomc.chromatic.com)

### CODE REVIEW

File: gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    @Override
+    public void configureEntrypoints(Map<String, EntrypointConnectorPlugin<?, ?>> entrypoints) {
+        entrypoints.putIfAbsent("http-get", EntrypointBuilder.build("http-get", HttpGetEntrypointConnectorFactory.class));
+    }
+
+    @Test
+    void should_receive_all_messages(HttpClient client, Vertx vertx) {
+        // In order to simplify the test, Kafka endpoint's consumer is configured with "autoOffsetReset": "earliest"
+        // It allows us to publish the messages in the topic before opening the api connection through SSE entrypoint.
+        Single.fromCallable(() -> getKafkaProducer(vertx)).flatMapCompletable(producer -> publishMessage(producer, "message1").andThen(publishMessage(producer, "message2")).andThen(publishMessage(producer, "message3")).doFinally(producer::close)).blockingAwait();
```
Comment (benoitgravitee) [inline gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java]: maybe add lines after dots for readability

File: gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            return request.send();
+        }).flatMap(response -> {
+            assertThat(response.statusCode()).isEqualTo(200);
+            return response.body();
+        }).test().awaitDone(30, TimeUnit.SECONDS).assertValue(body -> {
+            final JsonObject jsonResponse = new JsonObject(body.toString());
+            final JsonArray items = jsonResponse.getJsonArray("items");
+            assertThat(items).hasSize(2);
+            final JsonObject message = items.getJsonObject(0);
+            assertThat(message.getString("content")).isEqualTo("message1");
```
Comment (benoitgravitee) [inline gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java:82]: you should test what the message should not contain.

File: gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+    @Override
+    public void configureEntrypoints(Map<String, EntrypointConnectorPlugin<?, ?>> entrypoints) {
+        entrypoints.putIfAbsent("http-get", EntrypointBuilder.build("http-get", HttpGetEntrypointConnectorFactory.class));
+    }
+
+    @Test
+    void should_receive_all_messages(HttpClient client, Vertx vertx) {
+        // In order to simplify the test, Kafka endpoint's consumer is configured with "autoOffsetReset": "earliest"
+        // It allows us to publish the messages in the topic before opening the api connection through SSE entrypoint.
```
Comment (benoitgravitee) [inline gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java]: typo on comment here and below (remove SSE add HTTP-GET)

File: gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+        }).flatMap(response -> {
+            assertThat(response.statusCode()).isEqualTo(200);
+            return response.body();
+        }).test().awaitDone(30, TimeUnit.SECONDS).assertValue(body -> {
+            final JsonObject jsonResponse = new JsonObject(body.toString());
+            final JsonArray items = jsonResponse.getJsonArray("items");
+            assertThat(items).hasSize(1);
+            final JsonObject message3 = items.getJsonObject(0);
+            assertThat(message3.getString("content")).isEqualTo("message3");
+            assertThat(message3.getString("id")).isEqualTo("dGVzdC10b3BpY0AwIzI="); // test-topic@0#1
```
Comment (benoitgravitee) [inline gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointKafkaEndpointIntegrationTest.java]: there is mismatch between base64 and the comment, may I suggest you use a utility method to convert on the fly (might help reading)

File: gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointMqtt5EndpointIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                    request.putHeader(HttpHeaderNames.ACCEPT.toString(), MediaType.APPLICATION_JSON);
+                    return request.send();
+                })
+                .doOnSuccess(response -> assertThat(response.statusCode()).isEqualTo(400))
+                .flatMap(HttpClientResponse::body)
+                .test()
+                .awaitDone(30, TimeUnit.SECONDS)
+                .assertValue(body -> {
+                    final JsonObject jsonResponse = new JsonObject(body.toString());
+                    assertThat(jsonResponse.getString("message")).isEqualTo("Incompatible Qos capabilities between entrypoint requirements and endpoint supports");
```
Comment (benoitgravitee) [inline gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointMqtt5EndpointIntegrationTest.java:98]: test that you don't have id (and other fields ?)

Comment (guillaumelamirand) [reply gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointMqtt5EndpointIntegrationTest.java:98]: Why should have receive an id ? this is not a message payload but an error payload.

**Review by ytvnr — APPROVED** [review]

**Pull request merged**